Doc/library/types.rst

```diff
@@ -481,14 +481,25 @@ Additional Utility Classes and Functions
    A simple :class:`object` subclass that provides attribute access to its
    namespace, as well as a meaningful repr.
 
-   Unlike :class:`object`, with ``SimpleNamespace`` you can add and remove
-   attributes.  If a ``SimpleNamespace`` object is initialized with keyword
-   arguments, those are directly added to the underlying namespace.
+   Unlike :class:`object`, with :class:`!SimpleNamespace` you can add and remove
+   attributes.
+
+   :py:class:`SimpleNamespace` objects may be initialized
+   in the same way as :class:`dict`: either with keyword arguments,
+   with a single positional argument, or with both.
+   When initialized with keyword arguments,
+   those are directly added to the underlying namespace.
+   Alternatively, when initialized with a positional argument,
+   the underlying namespace will be updated with key-value pairs
+   from that argument (either a mapping object or
+   an :term:`iterable` object producing key-value pairs).
+   All such keys must be strings.
 
    The type is roughly equivalent to the following code::
 
        class SimpleNamespace:
-           def __init__(self, /, **kwargs):
+           def __init__(self, mapping_or_iterable=(), /, **kwargs):
+               self.__dict__.update(mapping_or_iterable)
                self.__dict__.update(kwargs)
 
            def __repr__(self):
```

Comment:
It may be outside the scope of this PR, but I noticed that in gh-109175 you didn't add `__replace__()` to the "equivalent" code here.  It's probably fine that this isn't an exact equivalence, especially since we say "roughly equivalent".

I just wanted to mention it (before I forget), in case it's something that would make sense to address in this PR.  I expect it would basically be what you wrote in https://github.com/python/cpython/issues/109174#issuecomment-1712430549.

Either way, I'll leave it up to you.

Reply:
It also does not have `__reduce__()`. Both methods would be simple, but I am not sure that it is needed to add them in this example. They may shadow the main idea of the class.